### PR TITLE
FSE: Add header logo placeholder

### LINKED ISSUE #35124: FSE: Add a logo image block placeholder to Modern business header
Feedback from Aadil on the flows so far brought up the idea of including an image placeholder in the header for a logo.

This is likely to be a common enough occurrence that we should look at this before launch.

We do have a medium term plan to include alternative header section templates that would include a logo, but the default could still have one.

An issue I see with this -- the image placeholder is too big and doesn't look like "spot for a logo". We should try to target this image block and reduce the placeholder size to something more like this:

Screen Shot 2019-08-03 at 12 06 44 PM

### PR DESCRIPTION
**Pull request by glendaviesnz**
**Important Note** This PR currently depends on timing of merge of https://github.com/Automattic/themes/pull/1167/files - this will only currently work if that PR is not merged after - will need some modification if merged after that.

#### Changes proposed in this Pull Request

* Add class to FSE site logo placeholder in order to allow targeting of CSS to placeholder image

#### Testing instructions

* Check out branch for https://github.com/Automattic/themes/pull/1197, build Modern Business and copy to FSE test environment
* Build this branch for FSE test environment
* Remove existing FSE templates, remove modern-business-fse-template-data flag and then re-activate modern business theme.
* Check that the Header template includes a site logo placeholder, and that when a image is added it centre aligns in the front end.

**Note:** as specified in #35124 the image placeholder should display narrower than the default one - currently fixed to 380px. Only the placeholder image is the reduced size, I have kept the actually block sizing the same to avoid having to test all the edge cases of different image sizes if we played with the actual block width.

**Header in editor after patch applied and templates rebuilt**

Screen Shot 2019-08-06 at 10 26 02 AM

**Front end view**
Screen Shot 2019-08-06 at 10 30 37 AM

Fixes #35124

### CODE REVIEW

**Comment by gwwar** [comment]
As a note https://github.com/Automattic/themes/pull/1167 already landed, and shouldn't be a blocker for this piece.

**Comment by gwwar** [comment]
With D31262-code landed, @glendaviesnz did you update content on the demo site? I'm seeing the following from the API response:

Screen Shot 2019-08-06 at 3 42 32 PM

**Comment by glendaviesnz** [comment]
> With [D31262-code](https://href.li/?https://code.a8c.com/D31262) landed, @glendaviesnz did you update content on the demo site? I'm seeing the following from the API response:
> 
> Screen Shot 2019-08-06 at 3 42 32 PM

Will get this added this afternoon and then tidy up the two PRs for this

**Comment by glendaviesnz** [comment]
The header image data has now been added to modern-business template api so this can be tested now.

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/site-logo.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * WordPress dependencies
+ */
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { addFilter } from '@wordpress/hooks';
+
+/**
```
Comment (vindl) [inline apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/site-logo.js]: We don't need this comment block if there are no internal dependencies afaik.

Comment (Copons) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/site-logo.js]: The WP dependencies block doesn't clear the ESLint import rule.
In other cases (back in the Gutenlypso days) I've solved it by disabling the rule:
```
/* eslint-disable wpcalypso/import-docblock */
```

**Comment by shaunandrews** [comment]
> I'm wondering if it makes sense to display the full site-logo placeholder and buttons in template part blocks (for example, when editing a page), as opposed to something a little bit more toned-down in terms of UI elements that look like they should be interactive.

I had this same thought. It’s strange to see buttons that I can’t actually click.

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/site-logo.js (hunk 1 above)
Comment (glendaviesnz) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/site-logo.js]: the pre-commit lint rule is a bit broken for this, as Jacopo notes it is not happy with just the WP deps one, but have fixed it with his eslint-disable instead for now.

**Comment by glendaviesnz** [comment]
> I had this same thought. It’s strange to see buttons that I can’t actually click.

I have removed the image action buttons of not editing the template

**Comment by gwwar** [comment]
This tests well for me @glendaviesnz. Let's maybe land https://github.com/Automattic/wp-calypso/pull/35198 first, rebase this one and verify if new header changes look good. Note behavior in https://github.com/Automattic/wp-calypso/pull/35158#issuecomment-519234414
